### DISCUSSION TITLE
Forecasting with a exogenous variable

### DISCUSSION THREAD

**Discussion by Yasslight90** [microsoft/FLAML#465] [General]
Hi there,

I just upgraded my FLAML version to v0.9.7 to test the impact of a exogenous variable. To do so, I passed a dataframe of two columns (date, exogenous variable) to the X_train parameter and the series to forecast to the y_train parameter. However, the test failed and returned the following error: 

ValueError: Pandas data cast to numpy dtype of object. Check input data with np.asarray(data).

can you please give me an example to illustrate the use of FLAML with exogenous variables?

Thank you!

Yassine

**Comment by int-chaos** [comment]
Hey Yassine. Thanks for reaching out with your issue.

I'm not sure what your dataset looks like, but here are examples of using FLAML with exogenous variables: https://github.com/microsoft/FLAML/blob/main/test/automl/test_forecast.py#L161 and https://github.com/microsoft/FLAML/blob/main/test/automl/test_forecast.py#L242

Let me know if you have any other questions

**Comment by Yasslight90** [comment]
Thank you! It is all working fine now.
